test_integration/03_redeem_and_unstake/01_redeem_and_unstake.js

```diff
@@ -0,0 +1,269 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const BN = require('bn.js');
+const shared = require('../shared');
+const Utils = require('../../test/test_lib/utils');
+const EventDecoder = require('../../test/test_lib/event_decoder');
+const ProofUtils = require('../lib/proof_utils');
+const Anchor = require('../lib/anchor');
+const RedeemAssertion = require('./utils/redeem_assertion');
+const ConfirmRedeemIntentAssertion = require('./utils/confirm_redeem_intent_assertion');
+const ProgressRedeemAssertion = require('./utils/progress_redeem_assertion');
+const ProgressUnstakeAssertion = require('./utils/progress_unstake_assertion');
+const ProveGatewayAssertion = require('../lib/prove_gateway_assertion');
+
+describe('Redeem and Unstake', async () => {
+    let originAccounts;
+    let auxiliaryAccounts;
+    let originWeb3;
+    let auxiliaryWeb3;
+
+    let gateway;
+    let cogateway;
+    let token;
+    let baseToken;
+    let ostPrime;
+
+    let originAnchor;
+    let redeemRequest;
+
+    let redeemAssertion;
+    let progressRedeemAssertion;
+    let progressUnstakeAssertion;
+    let proofUtils;
+
+    before(async () => {
+        originWeb3 = shared.origin.web3;
+        auxiliaryWeb3 = shared.auxiliary.web3;
+        originAccounts = shared.origin.accounts;
+        auxiliaryAccounts = shared.origin.accounts;
+        token = shared.origin.contracts.Token;
+        baseToken = shared.origin.contracts.BaseToken;
+        gateway = shared.origin.contracts.EIP20Gateway;
+        cogateway = shared.auxiliary.contracts.EIP20CoGateway;
+        ostPrime = shared.auxiliary.contracts.OSTPrime;
+
+        const hasher = Utils.generateHashLock();
+
+        redeemRequest = {
+            amount: new BN(50),
+            gasPrice: new BN(1),
+            gasLimit: new BN(10),
+            redeemer: auxiliaryAccounts[2],
+            bounty: await cogateway.bounty.call(),
+            nonce: await cogateway.getNonce.call(auxiliaryAccounts[2]),
+            beneficiary: originAccounts[2],
+            hashLock: hasher.l,
+            unlockSecret: hasher.s,
+        };
+
+        originAnchor = new Anchor(
+            auxiliaryWeb3,
```

Comment:
`originAnchor` and `auxiliaryWeb3`? Does this pass right now? 😱

Reply:
O.o What's wrong here. 
Anchor uses auxiliaryWeb3 to get block and anchor state root to originAnchor.

Reply:
Ah, of course. This is contracts land where we don't need web3 to access origin 😅